Documentation/security/threat-model.rst

```diff
@@ -0,0 +1,751 @@
+
+For such an attacker, our threat matrix looks like:
+
+.. rst-class:: wrapped-table
+
++------------------+---------------------------------------------------+
+| **Threat         | **Identified STRIDE threats**                     |
+| surface**        |                                                   |
++==================+===================================================+
+| Cilium agent     | A Kubernetes API user with ``kubectl exec``       |
+|                  | access to the pod running Cilium effectively      |
+|                  | becomes a :ref:`root-equivalent host              |
+|                  | attacker <root-equivalent-host-attacker>`,        |
+|                  | since Cilium runs as a privileged pod.            |
+|                  | |br| |br|                                         |
+|                  | Users with permissions to configure workload      |
+|                  | settings could cause denial of service.           |
++------------------+---------------------------------------------------+
+| Cilium           | The ability to modify the ``Cilium*``             |
+| configuration    | CustomResourceDefinitions, as well as any         |
+|                  | CustomResource from Cilium, in the cluster could  |
+|                  | have the following effects:                       |
+|                  |                                                   |
+|                  | -  The ability to create or modify CiliumIdentity |
+|                  |    and CiliumEndpoint or CiliumEndpointSlice      |
+|                  |    resources would allow an attacker to tamper    |
+|                  |    with the identities of pods.                   |
+|                  | -  The ability to delete Kubernetes or Cilium     |
+|                  |    NetworkPolicies would remove policy            |
+|                  |    enforcement.                                   |
+|                  | -  Creating a large number of CiliumIdentity      |
+|                  |    resources could result in denial of service    |
+|                  |    service.                                       |
+|                  | -  Workloads external to the cluster could be     |
+|                  |    added to the network.                          |
+|                  | -  Traffic routing settings between workloads     |
+|                  |    could be modified                              |
+|                  |                                                   |
+|                  | The cumulative effect of such actions could       |
+|                  | result in the escalation of a single-node         |
+|                  | compromise into a multi-node compromise.          |
++------------------+---------------------------------------------------+
+| Cilium eBPF      | None                                              |
+| programs         |                                                   |
```

Comment:
Maybe this should also have a marker that indicates eBPF programs are also vulnerable for the `kubectl exec` case above, similar to my comments for the limited-privilege host attacker?